test/test-pages/gcba/sdk-controller.ts

```diff
@@ -0,0 +1,307 @@
+  const parentNode = document.getElementById('pending-sessions');
+  console.log('rendering pending sessions table', pendingSessions);
+  if (!pendingSessions.length) {
+    parentNode.innerHTML = '';
+    return;
+  }
+
+  let html = `<table class="table">
+    <thead>
+      <tr>
+        <th scope="col">conversationId</th>
+        <th scope="col">sessionId</th>
+        <th scope="col">autoAnswer</th>
+        <th scope="col">Answer</th>
+        <th scope="col">Decline</th>
+      </tr>
+    </thead>
+    <tbody>`;
+
+
+  pendingSessions.forEach(session => {
+    html += `<tr>
+    <th scope="row">${session.conversationId}</th>
+    <td>${session.id}</td>
+    <td>${session.autoAnswer}</td>
+    <td><button type="button" class="btn btn-success btn-sm" onclick="acceptPendingSession({conversationId:'${session.conversationId}'})"
+      >Answer</button>
+    </td>
+    <td><button type="button" class="btn btn-danger btn-sm" onclick="rejectPendingSession({conversationId:'${session.conversationId}'})"
+      >Decline</button>
+    </td>
+  </tr>`
+  });
+
+  html += `</tbody>
+  </table>`;
+
+  parentNode.innerHTML = html;
+}
+
+function renderSessions () {
+  const tableBodyId = 'session-tbody';
+  let tableBody = document.getElementById(tableBodyId);
+  let html = '';
+
+  if (!tableBody) {
+    const parentNode = document.getElementById('sessions-element');
+    parentNode.innerHTML = `<table class="table">
+      <thead>
+        <tr>
+          <th scope="col" scope="row">conversationId</th>
+          <th scope="col">sessionId</th>
+          <th scope="col">session state</th>
+        </tr>
+      </thead>
+      <tbody id="${tableBodyId}">
+      </tbody>
+    </table>`;
+    tableBody = document.getElementById(tableBodyId);
+  }
+
+  Object.values(conversationUpdatesToRender).forEach(session => {
+    const isSessionActive = session ? session.state === 'active' : undefined;
+
+    html += `<tr>
+    <th scope="row">${session.conversationId}</th>
+    <td>${session.id}</td>
+    <td class="${isSessionActive ? 'text-success' : 'text-danger'}">
+      ${session.state}
+    </td>
+  </tr>`
+  });
+  tableBody.innerHTML = html;
+}
+
+function cancelPendingSession (params) {
+  let output = `${_getLogHeader('cancelPendingSession')}
+    sessionId: ${params.sessionId}
+    conversationId: ${params.conversationId}`;
+
+  pendingSessions = pendingSessions.filter(s => s.conversationId !== params.conversationId);
+  renderPendingSessions();
+  writeToLog(output);
+}
+
+function handledPendingSession (params) {
+  let output = `${_getLogHeader('handledPendingSession')}
+    sessionId: ${params.sessionId}
+    conversationId: ${params.conversationId}`;
+
+  pendingSessions = pendingSessions.filter(s => s.conversationId !== params.conversationId);
+  renderPendingSessions();
+  writeToLog(output);
+}
+
+async function sessionStarted (session: IExtendedMediaSession) {
+  let output = `${_getLogHeader('sessionStarted')}
```

Comment:
Codacy has a fix for the issue: ['output' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
  const output = `${_getLogHeader('sessionStarted')}
```